### PR TITLE
[JENKINS-71414] Add DataBoundConstructor and Symbol to FolderConfigFileProperty

### PR DESCRIPTION
**Pull request by Dohbedoh**
…r overload

* Added DataboundConstructor to the `FolderConfigFileProperty`
* Added symbol `folderConfigFile` to the property descriptor
* Remove `owner` overload that is un-necessary 
* Deprecated constructor

### Testing done

* Spin up Jenkins
* Create a Folder
* Add Folder config files
* Save

--> Validate the the config files are still there after save
--> Validate the the config files are still there after restart

* create a pipeline job in the folder:

    ```
    node {
        configFileProvider([configFile(fileId: '<fileID>, targetLocation: 'test.txt', variable: 'TEST')]) {
          sh "cat $TEST"
        }
    }
    ```

* Run the pipeline

--> Validate that it properly injects the file and the content is expected

### Submitter checklist

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by alecharp** [comment]
Would it be possible to test this automatically?

File: src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileProperty.java [hunk 1]
```diff
@@ -116,6 +134,7 @@ public FolderConfigFileProperty reconfigure(StaplerRequest req, JSONObject form)
     }
 
     @Extension(optional = true)
+    @Symbol("folderConfigFile")
```
Comment (rsandell) [inline src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileProperty.java:137]:
```suggestion
    @Symbol("folderConfigFiles")
```

**Comment by rsandell** [comment]
Last time I checked JCasC didn't support creating folders and other Items, so the title isn't really correct.

File: src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileProperty.java [hunk 2]
```diff
@@ -6,31 +6,49 @@
...
-    private Collection<Config> configs = new TreeSet<>(COMPARATOR);
+    private Collection<Config> configs;
 
-    private transient AbstractFolder<?> owner;
+    /*package*/ FolderConfigFileProperty() {
+        this((Collection<Config>) null);
+    }
 
+    @Deprecated
     /*package*/ FolderConfigFileProperty(AbstractFolder<?> owner) {
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/configfiles/folder/FolderConfigFileProperty.java:36]: given this is package-private scope - should all references not be be updated and this removed rather then leaving deprecated as they are only in this code.

**Review by rsandell — APPROVED** [review]

**Review by rsandell — CHANGES_REQUESTED** [review]
Still approved just don't merge this yet.

**Review by rsandell — APPROVED** [review]

**Pull request merged**